### PR TITLE
enable Dependabot v2

### PR DESCRIPTION
**Pull request by sullis**
https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/

### CODE REVIEW

**Review by anuraaga — COMMENTED** [review]
I am personally anti-dependabot, updating dependencies multiple times between releases instead of once before a release is unproductive and spammy (security alerts without automatic PR spam is great of course). Will leave to other maintainers in case they have a different preference :)

**Comment by trask** [comment]
I have not used dependabot, so defer here to @anuraaga (and @iNikem if he has any preference).

**Comment by anuraaga** [comment]
Actually I realized in this repository in particular, we probably can't use dependabot anyways. We have a pattern of using compile only dependencies with old versions to allow instrumentation to work with the user's dependency versions. We have a small set of "real" dependencies but I guess dependabot doesn't support this use case.

**Comment by iNikem** [comment]
In general I am pro-Dependabot, but I also don't see how we can use it in this repo. We certainly cannot automatically upgrade library dependencies of the instrumentations.

File: .github/dependabot.yml
Comment (anuraaga) [inline .github/dependabot.yml:6]:
```suggestion
```

How about let's just leave it to checking the github actions versions, that seems pretty nice actually.

Comment (sullis) [reply .github/dependabot.yml]: > How about let's just leave it to checking the github actions versions, that seems pretty nice actually.

Done.

**Review by iNikem — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]
Thanks!

**Pull request merged**
